pkg/reduce/readloop/readloop.go

```diff
@@ -103,12 +103,13 @@ func (rl *ReadLoop) Startup(ctx context.Context) error {
 		// crosses the window.
 
 		alignedKeyedWindow := keyed.NewKeyedWindow(p.Start, p.End)
+
+		// insert the window to the list of active windows, since the active window list is in-memory
+		keyedWindow, _ := rl.windower.InsertIfNotPresent(alignedKeyedWindow)
+
 		// add key to the window, so that when a new message with the watermark greater than
 		// the window end time comes, key will not be lost and the windows will be closed as expected
-		alignedKeyedWindow.AddKey(p.Key)
-
-		// These windows have to be recreated as they are completely in-memory
-		rl.windower.CreateWindow(alignedKeyedWindow)
```

Comment:
this was a bug, while replaying partitions we used to create a new window for each partition
for ex: if we have `60-120-even` and `60-120-odd` partitions to be replayed, we used to create two separate windows 
but it should be one window(60-120) with two keys(even, odd)

I have updated the replay test to capture this case